### PR TITLE
Add `skip_rows` configuration and SkipRecordException

### PR DESCRIPTION
**Pull request by jca02266**
Added a feature to skip the record.

example of config.yml

```
filters:
  - type: ruby_proc
(snip)
    skip_rows:
      - proc: |
          ->(record) do
            if record["id"] == 13
              puts "skip! #{record.inspect}"
              true
            end
          end
(snip)
      - name: id
        proc: |
          ->(id) do
            if id == 14
              puts "skip! id==14"
              raise SkipRecordException
            end
            id * variables["multiply"]
          end
        type: string
(snip)
```

### CODE REVIEW

File: lib/embulk/filter/ruby_proc.rb [hunk 1]
```diff
@@ -175,17 +192,25 @@ def add(page)
           end
 
           record_hashes.each do |record_hash|
-            procs.each do |col, pr|
-              next unless record_hash.has_key?(col)
-              next if record_hash[col].nil? && skip_nils[col]
+            begin
+              skip_row_procs.each do |pr|
+                raise SkipRecordException if pr.call(record_hash)
```
Comment (joker1007) [inline lib/embulk/filter/ruby_proc.rb]: I think that this feature needs not to raise Exception.
It is non-local exit by condition.
To use throw-catch is better than to use Exception.

Especially in JRuby, Exception is high cost.
If non-local exit often occurs,  throw-catch is slightly fast.

**Comment by joker1007** [comment]
Thank you for your pull request !
I commented about exception usage.

File: lib/embulk/filter/ruby_proc.rb (hunk 1 above)
Comment (jca02266) [reply lib/embulk/filter/ruby_proc.rb]: Right. I rewrote the patch. I worried that thow-catch is rare feature in Ruby.

**Comment by jca02266** [comment]
Please review new patch.

**Comment by jca02266** [comment]
new example of config.yml

```
filters:
  - type: ruby_proc
(snip)
    skip_rows:
      - proc: |
          ->(record) do
            # You can write any conditions to discard a record in the skip_rows proc.
            if record["id"] == 13
              puts "skip! #{record.inspect}"
              true
            end
          end
(snip)
      - name: id
        proc: |
          ->(id) do
            if id == 14
              puts "skip! id==14"
              # If you want to discard a record in any columns proc, throw SkipRecord object
              throw SkipRecord
            end
            id * variables["multiply"]
          end
        type: string
(snip)
```

**Comment by jca02266** [comment]
Oops! CamelCase object is style violation. Do you like SKIP_RECORD, :skip_record or any other?

**Comment by joker1007** [comment]
I like `:skip_record`. I feel it is Rubyish.

**Comment by jca02266** [comment]
I changed to symbol.

**Comment by joker1007** [comment]
Thanks!

**Pull request merged**